aspnetcore/fundamentals/environments.md

```diff
@@ -165,7 +183,21 @@ The production environment should be configured to maximize security, performanc
 
 ## Set the environment
 
-It's often useful to set a specific environment for testing. If the environment isn't set, it defaults to `Production`, which disables most debugging features. The method for setting the environment depends on the operating system.
+It's often useful to set a specific environment for testing with an environment variable or platform setting. If the environment isn't set, it defaults to `Production`, which disables most debugging features. The method for setting the environment depends on the operating system.
+
+When the host is built, the last environment setting read by the app determines the app's environment. The app's environment can't be changed while the app is running.
+
+::: moniker range=">= aspnetcore-3.0"
+
+In addition to the following methods for setting the environment, the environment can also be set by calling <xref:Microsoft.Extensions.Hosting.HostingHostBuilderExtensions.UseEnvironment*> when building the host. See <xref:fundamentals/host/generic-host#environmentname>.
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-3.0"
+
+In addition to the following methods for setting the environment, the environment can also be set by calling <xref:Microsoft.AspNetCore.Hosting.HostingAbstractionsWebHostBuilderExtensions.UseEnvironment*> when building the host. See <xref:fundamentals/host/web-host#environment>.
```

Comment:
This is the worst of everything - sending them off to programatically setting the env. I see why that's not preferred. Why give them the worst option first?

This needs to come last, not first. And then something like

Although not recommended, UseEnvironment can set the env programmatically. FMI see

Reply:
That's what I said ... I wanted to put it at the end in its own section.

Can't add a single line *in the last subsection of this section*. It has to go in its own section at the end, which is what I originally proposed.